app/src/main/java/com/example/plantsapp/presentation/ui/tasks/TasksFragment.kt

```diff
@@ -39,4 +42,23 @@ class TasksFragment : Fragment(R.layout.fragment_tasks) {
             }
         }
     }
+
+    companion object {
+        private const val ARGUMENT_DATE = "DATE"
+        private var Bundle.date: Date
+        get() = Date(
+            getLong(ARGUMENT_DATE)
+        )
+        set(date) {
+            putLong(ARGUMENT_DATE, date.time)
+        }
+
+        fun newInstance(date: Date): TasksFragment {
+            val tasksFragment = TasksFragment()
+            val args = bundleOf()
```

Comment:
You can use something like
tasksFragment.arguments = with(Bundle()){
   this.date = date
}